api/api.py

```diff
@@ -758,6 +758,9 @@ def buy_listing(user, key):
     if not listing.started or listing.ended:
         return jsonify({'message': "Listing not active."}), 403
 
+    if m.Sale.query.filter_by(listing_id=listing.id, buyer_id=user.id, state=m.SaleState.REQUESTED.value).first():
+        return jsonify({'message': "You already have an active purchase for this listing."}), 403
```

Comment:
Confirm me if this is because it shouldn't happen because the user will not have a "buy" button to click?

Reply:
Yes, I confirm that the "buy" button is not visible - all you can see is the QR code.

This code is to protect from somebody hitting the API directly.